src/asound/bluealsa-pcm.c

```diff
@@ -279,6 +279,15 @@ static int bluealsa_stop(snd_pcm_ioplug_t *io) {
 	struct bluealsa_pcm *pcm = io->private_data;
 	debug2("Stopping");
 
+       /* It is possible for this function to be called from an invalid state,
```

Comment:
I'm not sure whether it resolves problem or just covers it :). If state is not STATE_RUNNING (or similar), then IO thread should not be stared as well, so the `pthread_cancel()` will not be called (the `if` below). So, the only function which causes trouble is `bluealsa_dbus_pcm_ctrl_send_drop()`, am I correct? If so, then the problem should be fixed inside the bluez-alsa code. The "grand design" of bluez-alsa is that bluealsa server handles everything correctly and the ALSA plugin is just a wrapper for ALSA API compatibility :D.